NitroxPatcher/Patches/Dynamic/PrisonPredatorSwimToPlayer_Perform_Patch.cs

```diff
@@ -0,0 +1,49 @@
+using System.Collections.Generic;
+using System.Reflection;
+using System.Reflection.Emit;
+using HarmonyLib;
+using NitroxModel.DataStructures;
+using NitroxModel.Helper;
+using UnityEngine;
+
+namespace NitroxPatcher.Patches.Dynamic;
+
+public sealed partial class PrisonPredatorSwimToPlayer_Perform_Patch : NitroxPatch, IDynamicPatch
+{
+    internal static readonly MethodInfo TARGET_METHOD = Reflect.Method((PrisonPredatorSwimToPlayer t) => t.Perform(default, default, default));
+
+    public static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions)
+    {
+        // Replace everything with
+        // return EvaluatePriority(__instance, creature, time);
+        yield return new CodeInstruction(OpCodes.Ldarg_0);
+        yield return new CodeInstruction(OpCodes.Ldarg_1);
+        yield return new CodeInstruction(OpCodes.Ldarg_2);
+        yield return new CodeInstruction(OpCodes.Call, Reflect.Method(() => Perform(default, default, default)));
+        yield return new CodeInstruction(OpCodes.Ret);
+    }
+
+    public static void Perform(PrisonPredatorSwimToPlayer instance, Creature creature, float time)
```

Comment:
Is there a reason we have to implement these methods ourselves and can't hook into the current returns/logic?

Reply:
I added the explanation along with the reviews you made, tell me if it's clear enough

Reply:
Yeah that's a good explanation, nice work